src/types/sep8.ts

```diff
@@ -0,0 +1,47 @@
+import { ApprovalResponseStatus } from "../constants/sep8";
+
+export interface ApprovalResponse {
+  status:
+    | ApprovalResponseStatus.success
+    | ApprovalResponseStatus.revised
+    | ApprovalResponseStatus.pending
+    | ApprovalResponseStatus.actionRequired
+    | ApprovalResponseStatus.rejected;
+}
+
+export interface TransactionApproved extends ApprovalResponse {
```

Comment:
I don't think we need to extend `ApprovalResponse` for these. There is only one attribute there at the moment and we set it for every extended type separately. We might keep them if there will be more shared attributes in the future.

Reply:
Got it. On the same topic, is it a common pattern to have the return type of a API call to be the base response, i.e. `ApprovalResponse` in this case, and depends on the `status`, we marshal the response to the associated extended type?